homeassistant/components/myuplink/config_flow.py

```diff
@@ -46,8 +49,17 @@ async def async_step_reauth_confirm(
 
     async def async_oauth_create_entry(self, data: dict) -> ConfigFlowResult:
         """Create or update the config entry."""
+
+        token = jwt.decode(
+            data["token"]["access_token"], options={"verify_signature": False}
+        )
+        uid = token["sub"]
+        await self.async_set_unique_id(uid)
+
         if self.source == SOURCE_REAUTH:
+            self._abort_if_unique_id_mismatch(reason="reauth_account_mismatch")
```

Comment:
Let's just call this `account_mismatch`, this way we can use the same string if we ever opt for implementing reconfigure :)